sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -317,32 +428,54 @@ def _environments_query(
             return query.lock(copy=False)
         return query
 
+    def _hydrate_seeds(self, snapshot_ids: t.Optional[t.Iterable[SnapshotIdLike]] = None) -> None:
+        query = (
+            exp.select("name", "identifier", "content")
+            .from_(self.seeds_table)
+            .where(None if not snapshot_ids else self._snapshot_id_filter(snapshot_ids))
+        )
+        for name, identifier, content in self.engine_adapter.fetchall(query):
+            snapshot = self._snapshot_cache.get(SnapshotId(name=name, identifier=identifier))
+            if snapshot and not snapshot.seed_is_hydrated:
+                assert isinstance(snapshot.model, SeedModel), "Must be seed model to hydrate"
+                snapshot.model = snapshot.model.to_hydrated(content)
+
     def _get_snapshots(
         self,
         snapshot_ids: t.Optional[t.Iterable[SnapshotIdLike]] = None,
         lock_for_update: bool = False,
         hydrate_seeds: bool = False,
+        hydrate_intervals: bool = True,
+        intervals_by_id: bool = False,
     ) -> t.Dict[SnapshotId, Snapshot]:
         """Fetches specified snapshots or all snapshots.
 
         Args:
             snapshot_ids: The collection of snapshot like objects to fetch.
             lock_for_update: Lock the snapshot rows for future update
             hydrate_seeds: Whether to hydrate seed snapshots with the content.
-
+            hydrate_intervals: Whether to hydrate interval snapshots.
         Returns:
             A dictionary of snapshot ids to snapshots for ones that could be found.
         """
+        snapshot_ids = list(snapshot_ids) if snapshot_ids else []
+        if not snapshot_ids:
+            return {}
+        if self._snapshot_cache and self._snapshot_cache.contains(snapshot_ids or set()):
```

Comment:
Why is it all or nothing? Can some IDs be present in the cache but not the others?